search/utils.py

```diff
@@ -33,104 +33,132 @@ def index_indicator(self, indicator):
+            except RequestError:
+                print(wf, "Error")
 
-    def delete_workflowlevel1(self, id):
+    def delete_workflowlevel1(self, wf):
         if self.es is None:
             return
 
         try:
-            self.es.delete(index=self.prefix+"workflow_level1", id=id, doc_type='workflow')
+            if wf.organization is not None and wf.organization is not None:
+                org_uuid = str(wf.organization.organization_uuid)
+
+                self.es.delete(index=self.prefix + org_uuid + "_workflow_level1", id=wf.level1_uuid, doc_type='workflow')
         except RequestError:
             print(id, "Error")   # Todo write to error log
 
-    def delete_workflowlevel2(self, id):
+    def delete_workflowlevel2(self, wf):
         if self.es is None:
             return
 
         try:
-            self.es.delete(index=self.prefix+"workflow_level2", id=id, doc_type='workflow')
+            if wf.workflowlevel1.organization is not None and wf.workflowlevel1.organization is not None:
+                org_uuid = str(wf.workflowlevel1.organization.organization_uuid)
+
+                self.es.delete(index=self.prefix + org_uuid + "_workflow_level2", id=wf.level2_uuid, doc_type='workflow')
         except RequestError:
             print(id, "Error")   # Todo write to error log
 
     def index_collecteddata(self, d):
         if self.es is None:
             return
 
-        # get model field data
-        data = self.get_field_data(d)
+        if d.workflowlevel1 is not None and d.workflowlevel1.organization is not None:
+            org_uuid = str(d.workflowlevel1.organization.organization_uuid)
 
-        # aggregate related models
-        di = d.indicator
-        data['indicator'] = di.name
+            # get model field data
+            data = self.get_field_data(d)
 
-        # index data with elasticsearch
-        try:
-            self.es.index(index=self.prefix+"collected_data", id=data['data_uuid'], doc_type='data_collection', body=data)
-        except RequestError:
-            print(d, "Error")   # Todo write to error log
+            # aggregate related models
+            di = d.indicator
+            data['indicator'] = di.name
 
-    def delete_collecteddata(self, id):
+            # index data with elasticsearch
+            try:
+                self.es.index(index=self.prefix + org_uuid + "_collected_data", id=data['data_uuid'], doc_type='data_collection', body=data)
+            except RequestError:
+                print(d, "Error")   # Todo write to error log
+
+    def delete_collecteddata(self, d):
         if self.es is None:
             return
 
         try:
-            self.es.delete(index=self.prefix+"collected_data", id=id, doc_type='data_collection')
+            if d.workflowlevel1.organization is not None and d.workflowlevel1.organization is not None:
```

Comment:
Yeah, I'm sure it won't be None! 😂

Reply:
As you're checking if the element is not `None`, you can do like: `if d.workflowlevel1.organization`